pkg/volume/portworx/portworx_util.go

```diff
@@ -330,3 +335,14 @@ func (util *PortworxVolumeUtil) getPortworxDriver(volumeHost volume.VolumeHost,
 
 	return volumeclient.VolumeDriver(util.portworxClient), nil
 }
+
+// lookupPXAPIPortFromService goes over all the ports in the given service and returns the target
+// port for osdMgmtDefaultPort
+func lookupPXAPIPortFromService(svc *v1.Service) int32 {
+	for _, p := range svc.Spec.Ports {
+		if p.Port == osdMgmtDefaultPort {
+			return p.TargetPort.IntVal
```

Comment:
What if the service uses named port?

Reply:
Portworx users use a fixed portworx-service spec generated from https://install.portworx.com. Hence above logic that looks for `port` and `targetPort` works.

The spec is 

```
kind: Service
apiVersion: v1
metadata:
  name: portworx-service
  namespace: kube-system
  labels:
    name: portworx
spec:
  selector:
    name: portworx
  ports:
    - name: px-api
      protocol: TCP
      port: 9001
      targetPort: 9001
```